### PR TITLE
Installing XRT 2021.1 packages in docker container

### PR DESCRIPTION
**Pull request by SubhedarV**
- added Docker file for installing packages in docker container 
- added steps to copy packages in Docker file dir and added docker build command for installing packages

### CODE REVIEW

File: .github/workflows/xrt_master_2021.1.yml [hunk 1]
```diff
@@ -212,6 +219,95 @@ jobs:
...
+          echo "${{ secrets.HARBOR_PASSWORD }}" | docker login registry.amd.com -u '${{ secrets.HARBOR_USERNAME }}' --password-stdin        
+        shell: bash          
+        
+      - name: Push Docker images to AMD Harbor          
+        run: |          
+          for package_file in "${{ github.workspace }}/${{ env.XRT_VERSION_PATCH }}/Dockerfile/"*-xrt.rpm "${{ github.workspace }}/${{ env.XRT_VERSION_PATCH }}/Dockerfile/"*-xrt.deb; do          
+            image_tag=$(basename "$package_file" | sed 's/\.[^.]*$//')          
+            echo "Pushing image xrt_image:$image_tag"          
+            docker tag "xrt_image:$image_tag" "registry.amd.com/buildops/xrt_image:$image_tag"          
+            docker push "registry.amd.com/buildops/xrt_image:$image_tag"
```
Comment (sunilvenkatraman-amd) [inline .github/workflows/xrt_master_2021.1.yml:306]: Why are we pushing to buildops? Can we create a new registry for XRT images?

File: .github/workflows/xrt_master_2021.1.yml [hunk 2]
```diff
@@ -212,6 +219,95 @@ jobs:
           USER: ${{ secrets.USER }}
           github-server-url: ${{ secrets.SERVER_URL }}
 
+      - name: Find and copy packages      
+        run: |      
+          cd ${{ env.PATH_DIR }}/artifact
```
Comment (sunilvenkatraman-amd) [inline .github/workflows/xrt_master_2021.1.yml:224]: Could you move this entire block into a separate script. Would really help with readability

File: .github/workflows/xrt_master_2021.1.yml [hunk 3]
```diff
@@ -212,6 +219,95 @@ jobs:
...
+        
+      - name: List contents of Dockerfile directory      
+        run: |      
+          echo "Contents of Dockerfile directory:"      
+          ls -al ${{ github.workspace }}/${{ env.XRT_VERSION_PATCH }}/Dockerfile      
+        shell: bash       
+        
+      - name: Build Docker images          
+        run: |          
+          set -e  # Exit on any error
```
Comment (sunilvenkatraman-amd) [inline .github/workflows/xrt_master_2021.1.yml:257]: Same comment as above. This is a shell script and we should probably have it separated. Would make it:
1. More readable
2. easily callable/testable outside of github actions

File: .github/workflows/xrt_master_2021.1.yml [hunk 4]
```diff
@@ -212,6 +219,95 @@ jobs:
...
+              elif [[ "$package_file" == *"8.1.1911"* ]]; then            
+                  base_image="centos:8"            
+                  install_commands='sed -i "s|^mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-* && sed -i "s|^#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g" /etc/yum.repos.d/CentOS-* && yum install -y epel-release && yum install -y dnf-plugins-core wget && yum config-manager --set-enabled powertools && yum clean all && yum install -y /tmp/${PACKAGE_FILE}  /tmp/xrm_202220.1.5.103_8.1.1911-x86_64.rpm && rm -f /tmp/${PACKAGE_FILE} /tmp/xrm_202220.1.5.103_8.1.1911-x86_64.rpm'  
+                  # install_commands='yum upgrade && yum install -y wget && yum install -y dnf-plugins-core wget && yum config-manager --set-enabled powertools && yum config-manager --set-enabled appstream && yum install -y /tmp/${PACKAGE_FILE} /tmp/xcr_202220.1.1.27_7.9.2009-x86_64.rpm /tmp/xrm_202220.1.5.103_7.8.2003-x86_64.rpm && rm -f /tmp/${PACKAGE_FILE} /tmp/xcr_202220.1.1.27_7.9.2009-x86_64.rpm /tmp/xrm_202220.1.5.103_7.8.2003-x86_64.rpm'
```
Comment (sunilvenkatraman-amd) [inline .github/workflows/xrt_master_2021.1.yml:281]: Please remove commented lines

File: .github/workflows/xrt_master_2021.1.yml [hunk 5]
```diff
@@ -212,6 +219,95 @@ jobs:
...
+                  # install_commands='sed -i "s|^mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-* && sed -i "s|^#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g" /etc/yum.repos.d/CentOS-* && yum install -y epel-release && yum install -y dnf-plugins-core wget && yum config-manager --set-enabled powertools && yum clean all && yum install -y /tmp/${PACKAGE_FILE} && rm -f /tmp/${PACKAGE_FILE}'  
+              elif [[ "$package_file" == *"amzn2"* ]]; then            
+                  base_image="amazonlinux:2"       
+                  install_commands='yum update -y && yum install -y wget gnupg2 && amazon-linux-extras install epel -y && yum install -y ocl-icd ocl-icd-devel && yum install -y tar && yum install -y /tmp/${PACKAGE_FILE} /tmp/xrm_202220.1.5.103_2-x86_64.rpm && rm -f /tmp/${PACKAGE_FILE} /tmp/xrm_202220.1.5.103_2-x86_64.rpm'
```
Comment (sunilvenkatraman-amd) [inline .github/workflows/xrt_master_2021.1.yml:285]: Why do we need ocl-icd ocl-icd-devel gnupg2?

File: .github/workflows/xrt_master_2021.1.yml (hunk 1 above)
Comment (SubhedarV) [reply .github/workflows/xrt_master_2021.1.yml]: Now pushing to new project - XRT

File: .github/workflows/xrt_master_2021.1.yml (hunk 3 above)
Comment (SubhedarV) [reply .github/workflows/xrt_master_2021.1.yml]: Updated code for addressing this comment - calling docker build images part from composite workflow

File: .github/workflows/xrt_master_2021.1.yml (hunk 4 above)
Comment (SubhedarV) [reply .github/workflows/xrt_master_2021.1.yml]: done

**Review by manikandan-xilinx — APPROVED** [review]

**Pull request merged**